prov/efa/src/efa_cq.h

```diff
@@ -0,0 +1,91 @@
+ * General Public License (GPL) Version 2, available from the file
+ * COPYING in the main directory of this source tree, or the
+ * BSD license below:
+ *
+ *     Redistribution and use in source and binary forms, with or
+ *     without modification, are permitted provided that the following
+ *     conditions are met:
+ *
+ *      - Redistributions of source code must retain the above
+ *        copyright notice, this list of conditions and the following
+ *        disclaimer.
+ *
+ *      - Redistributions in binary form must reproduce the above
+ *        copyright notice, this list of conditions and the following
+ *        disclaimer in the documentation and/or other materials
+ *        provided with the distribution.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+ * EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+ * MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+ * NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS
+ * BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN
+ * ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+ * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+#include "efa.h"
+
+/**
+ * @brief Create ibv_cq_ex by calling ibv_create_cq_ex
+ *
+ * @param[in] ibv_cq_init_attr_ex Pointer to ibv_cq_init_attr_ex
+ * @param[in] ibv_ctx Pointer to ibv_context
+ * @param[in,out] ibv_cq_ex Pointer to newly created ibv_cq_ex
+ * @return Return 0 on success, error code otherwise
+ */
+static inline int ibv_cq_ex_open_with_ibv_create_cq_ex(struct ibv_cq_init_attr_ex *ibv_cq_init_attr_ex,
+                                            struct ibv_context *ibv_ctx, struct ibv_cq_ex ***ibv_cq_ex)
+{
+
+	struct ibv_cq_ex *ibv_cq = ibv_create_cq_ex(ibv_ctx, ibv_cq_init_attr_ex);
+
+    if (ibv_cq) {
+        **ibv_cq_ex = ibv_cq;
+        return 0;
+    } else {
+        EFA_WARN(FI_LOG_CQ, "Unable to create extended CQ: %s\n", strerror(errno));
+        return -FI_ENOCQ;
+    }
+}
+
+/**
+ * @brief Create ibv_cq_ex by calling efadv_create_cq or ibv_create_cq_ex
+ *
+ * @param[in] ibv_cq_init_attr_ex Pointer to ibv_cq_init_attr_ex
+ * @param[in] efadv_cq_init_attr_ex Pointer to efadv_cq_init_attr_ex
+ * @param[in] ibv_ctx Pointer to ibv_context
+ * @param[in,out] ibv_cq_ex Pointer to newly created ibv_cq_ex
+ * @param[in,out] is_efadv_cq set to true if efadv_create_cq is used and false otherwise
+ * @return Return 0 on success, error code otherwise
+ */
+static inline int ibv_cq_ex_open(struct ibv_cq_init_attr_ex *ibv_cq_init_attr_ex,
+                                struct efadv_cq_init_attr *efadv_cq_init_attr,
+                                struct ibv_context *ibv_ctx, struct ibv_cq_ex **ibv_cq_ex,
+                                bool *is_efadv_cq)
+{
+    struct ibv_cq_ex *ibv_cq;
+    *is_efadv_cq = false;
+
+    #if HAVE_EFADV_CQ_EX
```

Comment:
The libfabric convention is to define separate function for each preprocessor block, i.e. you define different `ibv_cq_ex_open` for `HAVE_EFADV_CQ_EX` = 0 and 1

Reply:
ACK. Fixed.